src/windows/Installer.Windows/Installer.Windows.csproj

```diff
@@ -5,51 +5,29 @@
   <PropertyGroup>
     <TargetFramework>net472</TargetFramework>
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
-    <PayloadPath>$(PlatformOutPath)Payload.Windows\bin\$(Configuration)\net472\win-x86</PayloadPath>
     <EnableDefaultItems>false</EnableDefaultItems>
+    <PayloadPath>$(PlatformOutPath)Installer.Windows\bin\$(Configuration)\net472\win-x86</PayloadPath>
   </PropertyGroup>
 
   <ItemGroup>
     <None Include="Setup.iss" />
   </ItemGroup>
 
-  <ItemGroup>
-    <ProjectReference Include="../Payload.Windows/Payload.Windows.csproj" ReferenceOutputAssembly="false" />
-  </ItemGroup>
-
   <ItemGroup>
     <PackageReference Include="Tools.InnoSetup" Version="6.0.5" />
-    <PackageReference Include="Microsoft.VisualStudioEng.MicroBuild.Core" Version="1.0.0">
-      <PrivateAssets>all</PrivateAssets>
-    </PackageReference>
   </ItemGroup>
 
-  <!-- Since the installer file name includes the generated build number from Nerdbank.GitVersioning
-       we must create the FilesToSign item inside of a target that depends on GetBuildVersion and runs
-       before we attempt to sign any files or validate they exist. -->
-  <Target Name="CreateFilesToSignItems" DependsOnTargets="GetBuildVersion" BeforeTargets="PrepareForRun">
-    <ItemGroup>
-      <FilesToSign Include="$(OutDir)gcmcore*-win-x86-$(BuildVersionSimple).exe">
-        <Authenticode>Microsoft400</Authenticode>
-        <InProject>false</InProject>
-      </FilesToSign>
-    </ItemGroup>
-  </Target>
-
   <!-- Implicit SDK targets import (so we can override the default targets below) -->
   <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
 
-  <!-- Check all the files to sign exist -->
-  <Target Name="ValidateSigningDependencies" AfterTargets="PrepareForRun" Inputs="@(FilesToSign)" Outputs="$(OutDir)validatesign.timestamp">
-    <Error Text="File to sign not found: %(FilesToSign.Identity)" Condition="!Exists('%(FilesToSign.Identity)')" />
-    <WriteLinesToFile File="$(OutDir)validatesign.timestamp" Lines="@(FilesToSign)" Overwrite="true" />
-  </Target>
-
   <Target Name="CoreCompile" Condition="'$(OSPlatform)'=='windows'">
     <PropertyGroup>
       <InnoSetupCommandSystem>"$(NuGetPackageRoot)Tools.InnoSetup\6.0.5\tools\ISCC.exe" /DPayloadDir="$(PayloadPath)" /DInstallTarget=system "$(RepoSrcPath)\windows\Installer.Windows\Setup.iss" /O"$(OutputPath)"</InnoSetupCommandSystem>
       <InnoSetupCommandUser>"$(NuGetPackageRoot)Tools.InnoSetup\6.0.5\tools\ISCC.exe" /DPayloadDir="$(PayloadPath)" /DInstallTarget=user "$(RepoSrcPath)\windows\Installer.Windows\Setup.iss" /O"$(OutputPath)"</InnoSetupCommandUser>
     </PropertyGroup>
+
+    <Message Text="Lay Out" Importance="High" />
+    <Exec Command="powershell.exe –NonInteractive –ExecutionPolicy Unrestricted -Command &quot;&amp; {&amp;'$(MSBuildProjectDirectory)\layout.ps1' -Configuration '$(Configuration)' -Output '$(PayloadPath)'}&quot;" />
```

Comment:
<img width="617" alt="image" src="https://user-images.githubusercontent.com/5658207/170374899-4e8bc2d2-c65c-4e1e-86da-cca88fc67446.png">

The payload is signed, the installers are signed, but the installers now contain re-built/unsigned bits 😿 

The key with the Mac build is that we split up the build in to two parts:

1. `layout.sh`
2. `pack.sh`/`dist.sh`

The Installer.Mac MSBuild project calls a third script `build.sh` that calls `layout.sh && pack.sh && dist.sh`.

The release pipeline never relies on the `build.sh` script, but instead calls `layout.sh` (then signs on a Windows machine), then calls `pack.sh`/`dist.sh` directly.

I think we need to do the same thing here, but with a complication thanks to NuGet...

We need to create a new target inside of the `Installer.Windows` project called like `"BuildInstaller"`, that just calls the InnoSetup commands on a `$(PayloadPath)` that is assumed to be defined:

```xml
<Target Name="BuildInstaller">
  ..
  <Exec Command="$(InnoSetupCommandSystem)" />
  ...
</Target>
```

We then have the `Build` target call the `layout.ps1` script (as the `CoreCompile` target does now), but also then run `BuildInstaller` by setting `AfterTargets="Build"` on the "BuildInstaller" target:

```diff
- <Target Name="BuildInstaller">
+ <Target Name="BuildInstaller" AfterTargets="Build">
```

This means in a dev build, when `Installer.Windows` is built it will call `layout.ps1` and then after call the `BuildInstaller` target and run InnoSetup.

In our release pipeline we can then do this instead:

1. `dotnet build`
2. `dotnet test`
3. `.\src\windows\Installer.Windows\layout.ps1 -Output payload`
4. Sign files in `payload` to output to `signed-payload`
5. `dotnet msbuild src\windows\Installer.Windows -t:BuildInstaller -p:PayloadPath=signed-payload`
6. Sign the output InnoSetup installers produced from (5)
7. Upload `signed-payload` and output from (6)